gwdetchar/io/datafind.py

```diff
@@ -198,6 +198,8 @@ def get_data(channel, start, end, frametype=None, source=None,
                 'Could not determine observatory from frametype')
         except (HTTPError, JSONDecodeError):  # frame files not found
             pass
+    if len(source) == 0: # reset if no frames found
+        source = None
     if isinstance(source, list) and isinstance(channel, (list, tuple)):
```

Comment:
@alurban In order to prevent an empty list being passed, could we add an additional condition here?

```suggestion
    if source and isinstance(source, list) and isinstance(channel, (list, tuple)):
```

Reply:
Aha, this is indeed a necessary change, good catch @dethodav. The call to `remove_missing_channels` needs to stay (it prevents job failures when expected channels are not found in frames), so we need this change to line 201 and line 203 should read:

```python
    elif source:
```